src/FoxyClient.php

```diff
@@ -0,0 +1,637 @@
+<?php
+
+namespace Foxy\FoxyClient;
+
+/*
+
+config:
+    use_sandbox (boolean): set to true to connect to the API sandbox for testing.
+    access_token (string): pass in your properly scoped OAuth access_token to be used as a Authentication Bearer HTTP header.
+
+Examples:
+
+$psr17Factory = new \Nyholm\Psr7\Factory\Psr17Factory();
+$psr18Client = new \Buzz\Client\Curl($psr17Factory);
+
+//Get Homepage
+$fc = new FoxyClient($psr17Factory, $psr18Client);
+$result = $fc->get();
+die("<pre>" . print_r($result, 1) . "</pre>");
+
+//Get Authenticated Store
+$fc = new FoxyClient($psr17Factory, $psr18Client, [
+        'client_id' => $client_id,
+        'client_secret' => $client_secret,
+        'refresh_token' => $refresh_token,
+        'access_token' => $access_token, // optional
+        'access_token_expires' => $access_token_expires // optional
+    ]
+);
+$fc->get();
+$result = $fc->get($fc->getLink("fx:store"));
+die("<pre>" . print_r($result, 1) . "</pre>");
+
+Credits: David Hollander of https://foxytools.com/ got this project going and is awesome.
+*/
+
+use Exception;
+use Foxy\FoxyClient\Exceptions\JsonException;
+use Foxy\FoxyClient\Exceptions\ResponseException;
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class FoxyClient
+{
+    const PRODUCTION_API_HOME = 'https://api.foxycart.com';
+    const SANDBOX_API_HOME = 'https://api-sandbox.foxycart.com';
+    const LINK_RELATIONSHIPS_BASE_URI = 'https://api.foxycart.com/rels/';
+    const PRODUCTION_AUTHORIZATION_ENDPOINT = 'https://my.foxycart.com/authorize';
+    const SANDBOX_AUTHORIZATION_ENDPOINT = 'https://my-sandbox.foxycart.com/authorize';
```

Comment:
I guess it would be better if we move these static URL's to some kind of common config and reading the values from there to save some rework in future if there is chance of update in these endpoints.
Also if these endpoints are relative to an environment (eg. staging/sandbox/production) then it becomes more convenient and important to store them under some environment specific config.
@brettflorio